interpreter/interpreter.go

```diff
@@ -1011,16 +1017,17 @@ func resolveValue(v values.Value) (semantic.Node, error) {
 				return
 			}
 			var n semantic.Node
-			n, err = resolveValue(el)
+			n, ok, err := resolveValue(el)
 			if err != nil {
 				return
+			} else if ok {
+				node.Elements[i] = n.(semantic.Expression)
 			}
```

Comment:
I believe you need an else case here where `ok` is false and `err == nil`. The case should set a flag to false that the array can be in-lined. If any value of the array cannot be in-lined then the entire array cannot be in-lined.

Reply:
I'm not sure this is correct. If we see an array, then to my knowledge, it is being inlined. It's just one of the elements that can't be. So if you have something like:

```
msg = "message"
a = () => "foobar"
b = [msg, a()]
```

(Pretend that `a` cannot be resolved, even though in this case it is I think)

This would cause the `b` to be rewritten as `["foobar", a()]` which is still valid.

Reply:
The above code doesn't do that, it never adds anything to the array for values that cannot be resolved so you get `["foobar", nil]` with the current code.

Reply:
My mistake. I'll fix it.